File: 3309-count-prefix-and-suffix-pairs-i/count-prefix-and-suffix-pairs-i.js
/**
 * @param {string[]} words
 * @return {number}
 */
var countPrefixSuffixPairs = function (words) {
    let count = 0
    const isSuffix = (str1, str2) => {
        if (str1.length > str2.length) return false
        return str2.slice(-str1.length) === str1;
    }
    const isPrefix = (str1, str2) => {
        if (str1.length > str2.length) return false
        return str2.slice(0, str1.length) === str1;
    }
    for (let i = 0; i < words.length - 1; i++) {
        for (let j = i + 1; j < words.length; j++) {
            count += isPrefix(words[i], words[j]) && isSuffix(words[i], words[j])
        }
    }
    return count
};

File: 3309-count-prefix-and-suffix-pairs-i/count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        count = 0
        def is_suffix(str1, str2):
            if len(str1) > len(str2):
                return False
            return str2[-len(str1):] == str1

        def is_prefix(str1, str2):
            if len(str1) > len(str2):
                return False
            return str2[:len(str1)] == str1

        for i in range(len(words) - 1):
            for j in range(i + 1, len(words)):
                count += is_prefix(words[i], words[j]) and is_suffix(words[i], words[j])

        return count



